# Experiment 01

# Task 01

## Loading of Dataset and NumPy, Pandas Basics for Data Exploration

---

### Student Name

Karthik

### Course

B.Tech AIML

### Objective

To learn how to load datasets using NumPy and Pandas and perform basic data exploration.

## Step 1

### Import Required Libraries

NumPy is used for numerical computations.

Pandas is used for loading and analyzing datasets.

In [2]:
import numpy as np
import pandas as pd

## Step 2

# Creating a NumPy Array

NumPy (Numerical Python) is a Python library used for numerical computations.

A NumPy array stores multiple values of the same data type and allows mathematical operations to be performed efficiently.

In [3]:
# Creating a NumPy array

numbers = np.array([10, 20, 30, 40, 50])

print(numbers)

[10 20 30 40 50]


## Step 3

# Basic Operations on NumPy Arrays

NumPy provides built-in functions to perform statistical and mathematical operations on arrays.

In [4]:
print("Array :", numbers)

print("Maximum Value :", numbers.max())

print("Minimum Value :", numbers.min())

print("Sum :", numbers.sum())

print("Mean :", numbers.mean())

Array : [10 20 30 40 50]
Maximum Value : 50
Minimum Value : 10
Sum : 150
Mean : 30.0


## Step 4

# Introduction to Pandas DataFrame

A DataFrame is a two-dimensional data structure provided by the Pandas library.

It stores data in rows and columns, similar to an Excel spreadsheet or a database table.

DataFrames make it easy to organize, analyze, and manipulate datasets.

In [5]:
# Creating a dictionary

student_data = {
    "Name": ["Karthik", "Rahul", "Priya", "Anjali"],
    "Age": [20, 21, 20, 22],
    "Branch": ["AIML", "CSE", "ECE", "IT"],
    "Marks": [91, 85, 88, 95]
}

# Convert dictionary into DataFrame

df = pd.DataFrame(student_data)

# Display the DataFrame

df

,Name,Age,Branch,Marks
0,Karthik,20,AIML,91
1,Rahul,21,CSE,85
2,Priya,20,ECE,88
3,Anjali,22,IT,95


## Step 5

# Basic DataFrame Operations

Pandas provides several functions to understand the structure and contents of a dataset.

In [6]:
# Display first 2 rows

print(df.head(2))

print()

# Display last 2 rows

print(df.tail(2))

print()

# Shape of the DataFrame

print("Shape :", df.shape)

print()

# Column names

print("Columns :", df.columns)

      Name  Age Branch  Marks
0  Karthik   20   AIML     91
1    Rahul   21    CSE     85

     Name  Age Branch  Marks
2   Priya   20    ECE     88
3  Anjali   22     IT     95

Shape : (4, 4)

Columns : Index(['Name', 'Age', 'Branch', 'Marks'], dtype='object')


## Step 6

# Loading the Iris Dataset

The Iris dataset is one of the most popular datasets used in Machine Learning.

It contains measurements of iris flowers belonging to three different species.

The dataset contains:

- Sepal Length
- Sepal Width
- Petal Length
- Petal Width
- Species

In [7]:
# Load the Iris dataset

iris = pd.read_csv("iris.csv")

# Display the first five rows

iris.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


## Step 8

# Displaying the Last Five Rows

The `tail()` function displays the last five rows of the dataset. It helps verify the ending records and inspect the dataset.


In [8]:
iris.tail()


,sepal_length,sepal_width,petal_length,petal_width,species
145,6.7,3.0,5.2,2.3,virginica
146,6.3,2.5,5.0,1.9,virginica
147,6.5,3.0,5.2,2.0,virginica
148,6.2,3.4,5.4,2.3,virginica
149,5.9,3.0,5.1,1.8,virginica


## Step 9

# Finding the Shape of the Dataset

The `shape` attribute returns the number of rows and columns present in the dataset.

In [9]:
print("Shape of the dataset:", iris.shape)


Shape of the dataset: (150, 5)


## Step 10

# Displaying Column Names

The `columns` attribute displays the names of all columns in the dataset.

In [10]:
print(iris.columns)


Index(['sepal_length', 'sepal_width', 'petal_length', 'petal_width',
       'species'],
      dtype='object')


## Step 11

# Dataset Information

The `info()` function provides information about the dataset, including the number of rows, columns, data types, and missing values.

In [11]:
iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


## Step 12

# Statistical Summary

The `describe()` function generates descriptive statistics for numerical columns in the dataset.

In [12]:
iris.describe()

,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


## Step 13

# Checking Missing Values

The `isnull().sum()` function checks for missing values in each column of the dataset.

In [13]:
iris.isnull().sum()

,0
sepal_length,0
sepal_width,0
petal_length,0
petal_width,0
species,0


# Conclusion

Successfully loaded the Iris dataset using Pandas and performed basic exploratory data analysis.

The dataset contains 150 records and 5 columns with no missing values. Basic statistical measures and dataset structure were explored using Pandas functions.